bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NetatmoWelcomeBinding.java

```diff
@@ -0,0 +1,487 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.internal.welcome;
+
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.openhab.binding.netatmo.NetatmoBindingProvider;
+import org.openhab.binding.netatmo.internal.NetatmoException;
+import org.openhab.binding.netatmo.internal.authentication.OAuthCredentials;
+import org.openhab.binding.netatmo.internal.messages.NetatmoError;
+import org.openhab.binding.netatmo.internal.welcome.GetHomeDataResponse.Camera;
+import org.openhab.binding.netatmo.internal.welcome.GetHomeDataResponse.Event;
+import org.openhab.binding.netatmo.internal.welcome.GetHomeDataResponse.Home;
+import org.openhab.binding.netatmo.internal.welcome.GetHomeDataResponse.Person;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ *
+ * @author Ing. Peter Weiss
+ * @since 1.9.0
+ */
+public class NetatmoWelcomeBinding {
+
+    private static final Logger logger = LoggerFactory.getLogger(NetatmoWelcomeBinding.class);
+
+    private final Map<String, Map<String, Person>> configuredHomeKnownPersons = new HashMap<String, Map<String, Person>>();
+    private final Map<String, Map<String, Person>> configuredHomeUnKnownPersons = new HashMap<String, Map<String, Person>>();
+    private final Map<String, Home> configuredHomeMap = new HashMap<String, Home>();
+    private final Map<String, Map<String, Camera>> configuredhomeCameras = new HashMap<String, Map<String, Camera>>();
+
+    /**
+     * Execute the weather binding from Netatmo Binding Class
+     *
+     * @param oauthCredentials
+     * @param providers
+     * @param eventPublisher
+     */
+    public void execute(OAuthCredentials oauthCredentials, Collection<NetatmoBindingProvider> providers,
+            EventPublisher eventPublisher) {
+
+        logger.debug("Querying Netatmo Welcome API");
+        try {
+            oauthCredentials.refreshAccessToken();
+            configuredHomeKnownPersons.clear();
+            configuredHomeUnKnownPersons.clear();
+            configuredHomeMap.clear();
+            configuredhomeCameras.clear();
+
+            processGetHomeData(oauthCredentials, providers, eventPublisher);
```

Comment:
Should this happen every time? The weather binding only does this the first time.